src/lib/Imp.hs

```diff
@@ -614,12 +618,18 @@ makeDest allocInfo ty = do
           return $ Abs decls' $ AbsPtrs (Abs ptrBs' result) (map sink ptrInfo)
   runSubstReaderT idSubst $ translateDeclNest decls $ substM result
 
-makeDestRec :: forall n l.
-               Emits n => Idxs n -> IdxNest n l -> Type l -> DestM n (Dest n)
+type IdxNest = Nest Binder
+
+makeDestRec ::
+  forall n l . Emits n
+    => Idxs n       -- indices from the enclosing table lambas we're building
+    -> IdxNest n l  -- indices from the table type we're building a dest for.
```

Comment:
I'm not sure if I understand the distinction between those two arguments

Reply:
It was confusing to me too. My latest change changes the signature a bit and adds some more explanation.